api/datastore/postgres/postgres.go

```diff
@@ -562,3 +606,17 @@ func (ds *PostgresDatastore) Get(ctx context.Context, key []byte) ([]byte, error
 
 	return []byte(value), nil
 }
+
+
+func (ds *PostgresDatastore) Tx(f func(*sql.Tx) error) error {
+	tx, err := ds.db.Begin()
+	if err != nil {
+		return err
+	}
+	err = f(tx)
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+	return tx.Commit()
+}
```

Comment:
Does anyone have experience with the proper way to handle these transactions?

@seiflotfy and I are wondering if we need to account for the situation where ```f(tx)``` panics (L616).  If the process will terminate anyways, then we may not need to worry about an explicit ```Rollback()``` here.  On the other hand, if we expect something to recover higher up in the stack, then we should probably defer recover and take care of it.